programmer-skills/big-O-notation.md

```diff
@@ -0,0 +1,161 @@
+###Big-O notation explained  
+
+The Big-O notation is used to classify algorithms based on how their runtime grows relative to the input size, when the input gets arbitrarily large. 
+How much slower is the algorithm if the input has 10 000 items instead of 1?
+ 
+#####Formal Big-O notation definition from [wikipedia](https://en.wikipedia.org/wiki/Big_O_notation)
+ 
+>Let f and g be two functions defined on some subset of the real numbers. One writes  
+>f(x)=O(g(x)) as  x&rarr;&infin;,  
+>if and only if there is a positive constant M such that for all sufficiently large values of x, 
+>the absolute value of f(x) is at most M multiplied by the absolute value of g(x). That is, f(x) = O(g(x)) 
+>if and only if there exists a positive real number M and a real number x<sub>0</sub> such that  
+> |f(x)|&le;M|g(x)| for all x&ge; x<sub>0</sub>.
+ 
+ 
+Let's discuss the big-O notation through several examples
+
+###O(1)
+                                                          
+O(1) represents an algorithm that has the same runtime regardless of the size of the input data set.
+
+An easy example is when we check whether the first element of the array is null. The algorithm always checks 
+the first element in the array, regardless of the array's size.
+
+```javascript
+function isFirstItemNull(array)
+{
+	return array[0] === null;
+}
+```
+
+###O(N)
+
+O(N) describes an algorithm whose runtime will grow linearly and in direct proportion to the size of the input data set. For example  if the algorithm includes a single iteration
+ over the data set, the perfromance is O(N). 
+
+Our test algorithm checks whether an item is in the array or not.
+```javascript
+function containsItem(array, item){
+	for (var element in array)
```

Comment:
You should use for...of instead of the for...in loop here. Please see [here](https://github.com/codingforeveryone/READMEs/blob/master/JavaScript/ECMAScript%206.md#for-of-statement) for more info.